### DISCUSSION TITLE
Help with IOS notifications automation

### DISCUSSION THREAD

**Discussion by tismofied** [moralmunky/Home-Assistant-Mail-And-Packages#546] [Q&A] [answered]
I have the automation setup like this but I never get notifications at all even if my lovelace card shows mail and packages coming. I checked the trace and it says " Stopped because a condition failed at October 25, 2021, 8:36:07 AM (runtime: 0.00 seconds)" no logbook entries found under log
If I run the automation manually, I get the notification just fine

```  alias: Mail Notif - Mail Delieveries
  trigger:
  - platform: state
    entity_id: sensor.mail_usps_mail
  - platform: state
    entity_id: sensor.mail_packages_in_transit
  - platform: state
    entity_id: sensor.mail_amazon_packages
  condition:
  - condition: or
    conditions:
    - condition: template
      value_template: '{{ ((as_timestamp(now()) - as_timestamp(states.sensor.mail_usps_mail.last_changed))
        < 15 and states(''sensor.mail_usps_mail'') > 0) }}'
    - condition: template
      value_template: '{{ ((as_timestamp(now()) - as_timestamp(states.sensor.mail_packages_in_transit.last_changed))
        < 15 and states(''sensor.mail_packages_in_transit'') > 0) }}'
    - condition: template
      value_template: '{{ ((as_timestamp(now()) - as_timestamp(states.sensor.mail_amazon_packages.last_changed))
        < 15 and states(''sensor.mail_amazon_packages'') > 0) }}'
  action:
  - service: notify.mobile_app_iphone_12_pro_max
    data_template:
      title: '*Today''s Mail and Packages*'
      message: '{{ states(''sensor.mail_deliveries_message'')}}'
  - service: notify.mobile_app_iphone_12_pro_max
    data:
      title: Mail and Packages
      message: Here is Today's coming Mail
      data:
        attachment:
          url: '{{ states(''sensor.mail_image_url'') }}'
          hide-thumbnail: false
        apns_headers:
          apns-collapse-id: homeassistant-mail-update
        push:
          thread-id: homeassistant-mail
          sound:
            name: US-EN-Alexa-Mail-Has-Arrived.wav
            critical: 1
            volume: 0.3```

**Comment by firstof9** [comment]
Your quotes are wrong.

Wrong:
```yaml
      value_template: '{{ ((as_timestamp(now()) - as_timestamp(states.sensor.mail_usps_mail.last_changed))
        < 15 and states(''sensor.mail_usps_mail'') > 0) }}'
```
Correct:
```yaml
      value_template: "{{ ((as_timestamp(now()) - as_timestamp(states.sensor.mail_usps_mail.last_changed))
        < 15 and states('sensor.mail_usps_mail') > 0) }}"
```

**Reply by tismofied** [reply]
Thank you Chris for your help. Is that true for the action part  of the automation also or those quotes are correct?

**Reply by firstof9** [reply]
All your quotes need to alternate like that the outer most quotes cannot be the same as the inner most quotes.
Also you used `''` rather than `"`.

**Comment by tismofied** [comment]
Update: the notification came in today. Thank you Chris for your help again. The pic in the notification showed an error but I'll take it. 
at least it's working

**Reply by firstof9** [reply]
AFAIK the image needs to be available externally, there should be a checkbox option for this, that will allow the image to work properly.

**Reply by tismofied** [reply]
> AFAIK the image needs to be available externally, there should be a checkbox option for this, that will allow the image to work properly.

Is that option in the app? I can't find it

**Reply by firstof9** [reply]
It's in `Step 2` of the integration configuration:

![image](https://user-images.githubusercontent.com/1105672/139268282-079eb827-124c-458f-8f64-8f1063236f55.png)

**Reply by tismofied** [reply]
Yeah I have that checked. As well as the one on top of it. Still no image coming through.

**Reply by firstof9** [reply]
I'd double check [moralmunky's Wiki entry on iOS Notifications](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/wiki/Notifications#home-assistant-companion-app-ios-notification-example) then.

**Reply by tismofied** [reply]
will do Thanks

**Comment by tismofied** [comment]
I figured out the cause of the notification image error but I don't know why the cause is happening in the first place.
I fixed the quotes as mentioned above in the post but as soon as HA restarts it reverts back to the wrong quotes, why is that happening?
**Fixed quotes:**
[]
![Web capture_5-11-2021_93821_homeassistant local](https://user-images.githubusercontent.com/90809124/140519845-659987a6-ffd6-4b5c-939d-8d34fdd56800.jpeg)
 **Reverted wrong quotes after HA restart:**

![mail notif yaml](https://user-images.githubusercontent.com/90809124/140520009-e200efcc-a139-492d-88ac-cccf0b4f75ea.jpeg)

**Reply by firstof9** [reply]
Looks like you used the UI to make the automation, is that correct?

**Comment by tismofied** [comment]
Should I just delete it from the UI and write it in automation.yaml?

**Reply by firstof9** [reply]
I'd do that, the UI tries to help as you can see, it's breaking it for you.

**Reply by tismofied** [reply]
Ok I deleted it from HA. Restarted HA and added it to automations.yaml
Checked the config and came back good. 
Restarted HA again and went to automations and run action to test it then I get this 

![image](https://user-images.githubusercontent.com/90809124/140526301-5cadd2bd-8ccd-410d-84f2-4a34ca603c9a.jpeg)

Now I know it's not with my network because I fixed the config.yaml yesterday and run the action and the gif showed up on this notification on the phone. So I know it works

**Comment by firstof9** [comment, marked as answer]
Make sure `sensor.mail_image_url` shows the proper URL to get to your Home Assistant externally.

**Reply by tismofied** [reply]
That's it.  How do I edit the url? It's missing the port

**Reply by firstof9** [reply]
The URL is obtained from your Home Assistant configuration.
So either in your `configuration.yaml` or in the UI `Configuration -> General -> External URL`
![image](https://user-images.githubusercontent.com/1105672/140553088-0f35132f-ed08-47dd-b337-c04774ef1a14.png)
